### PR TITLE
Fixes #4642

### LINKED ISSUE #4642: Enhanced Stereo is lost when using GetMolFrags(m, asMols=True)
**Describe the bug**

When calling `GetMolFrags(m, asMols=True)` the returned molecules loose their enhanced stereo information (stereogroups) and instead get R/S markers IF there are multiple fragments in the molecule. If `m` is only one fragment, the enhanced stereo is preserved. 

I assume in that case the input is just returned unmodified where as with multiple components some form of copy is used that looses information.

**To Reproduce**

```
mx = Chem.MolFromSmiles('CCCC.C[C@H](F)C[C@H](O)Cl |o1:5,o2:8|')
components = Chem.GetMolFrags(mx, asMols=True)
components[1]
```

`components[1].GetStereoGroups()[0].GetGroupType()`

Will then also lead to an IndexError as no StereoGroups are actually present anymore.

**Expected behavior**

That the enhanced stereo is preserved.

**Screenshots**

image

**Configuration (please complete the following information):**
 - RDKit version:  2021.03.5
 - OS:  Win 10
 - Python version (if relevant): 3.9
 - Are you using conda? Yes
 - If you are using conda, which channel did you install the rdkit from? conda-forge

### PR DESCRIPTION
**Pull request by greglandrum**
Makes sure stereo groups survive the splitting. 

This potentially includes creating new stereo groups if one is split across fragments.

Here's a concrete example (from the discussion), C[C@](N)(O)F.C[C@@](F)(Cl)Br |o1:1,6,r|:
image

We know that we have:
image
or
image

So when we split them we know that the first fragment is:
image
or:
image

Which corresponds to:
image

So we should keep the stereogroup.

### CODE REVIEW

**Comment by greglandrum** [comment]
Tagging @ricrogz and @d-b-w because I know both of them have thought about stereogroups

**Comment by bp-kelley** [comment]
I think it could be useful to keep the stereo tags from split stereo groups: if the stereo wasn’t in a group it would be retained.

**Comment by bp-kelley** [comment]
Or do you consider it not specified now?

**Comment by greglandrum** [comment]
> I think it could be useful to keep the stereo tags from split stereo groups: if the stereo wasn’t in a group it would be retained.
> Or do you consider it not specified now?

That's the question.

Before getting into this too deeply: I don't anticipate that this is going to happen all that often, so we're probably discussing an edge case.

After thinking about it some more, I think we should actually keep the stereo groups in this case (with a different stereo group in each sub-molecule of course).

Here's a concrete example, `C[C@](N)(O)F.C[C@@](F)(Cl)Br |o1:1,6,r|`:
image

We know that we have:
image
or
image

So when we split them we know that the first fragment is:
image
or:
image

Which corresponds to:
image

**Comment by bp-kelley** [comment]
Just chatted with @jones-gareth and he has an rgroup decomp example that this might resolve.  I’ll work with him getting an example.

**Comment by greglandrum** [comment]
@bp-kelley @ricrogz this is ready for review now

File: Code/GraphMol/MolOps.cpp [hunk 1]
```diff
@@ -660,6 +659,29 @@ std::vector<ROMOL_SPTR> getMolFrags(const ROMol &mol, bool sanitizeFrags,
         (*fragsMolAtomMapping).push_back((*mci).second);
       }
     }
+    // copy stereoGroups (if present)
+    if (!mol.getStereoGroups().empty()) {
+      boost::dynamic_bitset<> keptSGs(mol.getStereoGroups().size(), 0);
```
[the remark this answers is not in the document]
Comment (greglandrum) [reply Code/GraphMol/MolOps.cpp:664]: nope, I will remove it

**Review by ricrogz — APPROVED** [review]

**Comment by d-b-w** [comment]
I agree with the new approach, yeah. Otherwise we're accidentally resolving stuff!

**Pull request merged**